distro.py

```diff
@@ -935,9 +935,23 @@ def _os_release_info(self):
         Returns:
             A dictionary containing all information items.
         """
-        if os.path.isfile(self.os_release_file):
-            with open(self.os_release_file) as release_file:
+        def _open_and_return_info(release_file_path):
+            """Simple closure to DRY below logic"""
+            with open(release_file_path) as release_file:
                 return self._parse_os_release_content(release_file)
+
+        # If any, try the os-release file specified at instantiation.
+        if self.os_release_file:
+            if os.path.isfile(self.os_release_file):
+                return _open_and_return_info(self.os_release_file)
+        else:
+            # If not, try known "regular" locations of os-release files.
+            for os_release_file_candidate in (
+                    os.path.join(self.etc_dir, _OS_RELEASE_BASENAME),
+                    os.path.join(self.usr_lib_dir, _OS_RELEASE_BASENAME)):
+                if os.path.isfile(os_release_file_candidate):
+                    return _open_and_return_info(os_release_file_candidate)
+
```

Comment:
While the current code checks for `os.path.isfile`, with regard to [the spec](https://www.freedesktop.org/software/systemd/man/os-release.html) it seems to me that we should respect a symlink with an existing target — no? Then `os.path.exists` might be the function of choice. What do you think?

PS: I think the DRY problem can be addressed in a simpler way. What about:

```python
for os_release_file in (
        self.os_release_file,
        os.path.join(self.etc_dir, _OS_RELEASE_BASENAME),
        os.path.join(self.usr_lib_dir, _OS_RELEASE_BASENAME),
        ):
    if os_release_file and os.path.isfile(os_release_file):
        with open(os_release_file) as f:
            return self._parse_os_release_content(f)

return {}
```

Reply:
I hadn't opted for this solution as by instantiating a `LinuxDistribution` object with a speficied `os_release_file` attribute, the developer _might_ expect the loading of their file **only**.
So the DRY method imposed itself, as the logic _has to_ (?) be separated in two distinct branches.

Reply:
I see how about:
```python
if self.os_release_file:
    candidates = [self.os_release_file]
else:
    candidates = [
        os.path.join(self.etc_dir, _OS_RELEASE_BASENAME),
        os.path.join(self.usr_lib_dir, _OS_RELEASE_BASENAME),
    ]

for os_release_file in candidates:
    if os_release_file and os.path.isfile(os_release_file):
        with open(os_release_file) as f:
            return self._parse_os_release_content(f)

return {}
```